### PR TITLE
feat(security): Eliminate security-secrets-setup module

### PR DESCRIPTION
**Pull request by jim-wang-intel**
After Vault is no longer running in TLS mode and thus no need for `security-secrets-setup` component to generate certificate for Vault.

Also user-specified Kong proxy certificates will be achieved through other utility like `secrets-config` tls command, thus all related codes will be no longer needed.
    
See issue #2867 for details.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [X ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?

`security-secrets-setup` codes and docker image built make still exists

## What is the new behavior?
`security-secrets-setup` codes and related docker built make are removed

## Does this PR introduce a breaking change?

- [ ] Yes
- [X ] No

## New Imports

- [ ] Yes
- [X ] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

- Depends on the parent [PR Remove Vault TLS](https://github.com/edgexfoundry/edgex-go/pull/2890) merged into master first.

- TBD PR:  The Consul related checkers on `secrets-setup` will also needs to be removed.  See issue: https://github.com/edgexfoundry/docker-edgex-consul/issues/41

## Other information
The current PR looks big changes but lots of them are due to the changes from the parent PR and file removals.  After the parent PR is merged into master, it should look smaller.

To test it one can use the local builds of edgex-go and edgex-consul to run in the docker-compose file (PR TBD- with `security-secrets-setup` services removed in developer-scripts).

### CODE REVIEW

File: .gitignore
Comment (jim-wang-intel) [inline .gitignore:22]: Note: this is a duplicate ignore as below in original line 31

**Review by lenny-intel — DISMISSED** [review]
LGTM, but needs to be rebased

**Comment by lenny-intel** [comment]
@jim-wang-intel,  Looks like the snap needs to be update to not expect this service to be there.

File: internal/security/secretstore/init.go
Comment (bnevis-i) [inline internal/security/secretstore/init.go:376]: Better check to see if this will generate a merge conflict with @beaufrusetta's change.  Probably better to get @beaufrusetta change in first, and see if this section still need to be updated.  Can follow up later with a PR to eliminate the cert uploading to Vault altogether.

Comment (beaufrusetta) [reply internal/security/secretstore/init.go:376]: This is definitely going to conflict, especially since I have a pending update to add an else statement with a log entry...and at the end of that block I have a closing log entry that states vault init complete.

Comment (lenny-intel) [reply internal/security/secretstore/init.go:376]: @beaufrusetta , we'll land your PR first and then @jim-wang-intel can rebase and resolve conflicts in his PR.

Comment (jim-wang-intel) [reply internal/security/secretstore/init.go]: rebase and resolved the conflicts, thanks guys.

File: snap/hooks/install
Comment (tonyespy) [inline snap/hooks/install:58]: I think these lines should be removed vs. commented out, unless there's some reason you think we should preserve them here?

File: snap/snapcraft.yaml
Comment (tonyespy) [inline snap/snapcraft.yaml:139]: Did your other PR to remove the vault consul backend land? If not, then I would leave consul here, and remove in the other PR.

File: snap/snapcraft.yaml
Comment (tonyespy) [inline snap/snapcraft.yaml:154]: What is this used for, a vault token? AFAIK there are no tokens written to the directory `$SNAP_DATA/config/security-secretstore-setup/res`.

File: snap/hooks/configure
Comment (tonyespy) [inline snap/hooks/configure:144]: Same comment as the install hook, I'd prefer you just remove this...

**Review by tonyespy — CHANGES_REQUESTED** [review]
Overall, this looks pretty good, and a quick snap build/run shows everything come up OK.

One question, with this change, does Kong auto-generate a self-signed TLS certificate, or does that require a configuration change?

Also re: the optimization patch, I'll help you re-generate as soon as you've got a clean `snapcraft.yaml`.

File: README.md
Comment (jim-wang-intel) [reply README.md:30]: yes, I can clean that up- just some note temporary.

File: README.md
Comment (jim-wang-intel) [reply README.md:34]: thanks, will re-wording to that you indicated

File: cmd/security-bootstrap-redis/README.md
Comment (jim-wang-intel) [reply cmd/security-bootstrap-redis/README.md:11]: i think the original author meant the `secretstore-setup` for storing the mongo credentials into Vault; that 's why i changed it.

File: snap/hooks/install
Comment (jim-wang-intel) [reply snap/hooks/install:58]: yeap, i'll remove it

File: snap/snapcraft.yaml
Comment (jim-wang-intel) [reply snap/snapcraft.yaml:139]: the other PR not land yet. I'll remove it after the other PR lands.

File: snap/snapcraft.yaml
Comment (jim-wang-intel) [reply snap/snapcraft.yaml:154]: well, that is used by toke file provider I believe. Since we are eliminating the `security-secrets-setup` and that path will be no more; so I change it to other folder path.  Note that in the original toml file it is set to `TokenFolderPath = "/vault/config/assets"`.    Should I change it to something else in $SNAP_DATA?

File: snap/hooks/configure
Comment (jim-wang-intel) [reply snap/hooks/configure:144]: sure, will remove it.

**Comment by jim-wang-intel** [comment]
> Overall, this looks pretty good, and a quick snap build/run shows everything come up OK.
> 
> One question, with this change, does Kong auto-generate a self-signed TLS certificate, or does that require a configuration change?
> 

@tonyespy  Yes, when Kong starts it will automatically generate an internal default TLS cert by itself.  One can use utility `./secrets-config proxy tls` subcommand to bring your own TLS certificate and upload to Kong.   See details for [uploading your own cert](https://github.com/edgexfoundry/edgex-go/tree/master/cmd/secrets-config#subcommands) if you choose to do so.

> Also re: the optimization patch, I'll help you re-generate as soon as you've got a clean `snapcraft.yaml`.

Ok, thanks for your help! I'll clean it up and let you know.

File: snap/snapcraft.yaml
Comment (jim-wang-intel) [reply snap/snapcraft.yaml]: this has been resovled

File: .gitignore
Comment (lenny-intel) [inline .gitignore:35]: Why add this? It shouldn't exist and if some change is made where we do want it, this will cause some issues.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (jim-wang-intel) [reply .gitignore:35]: it exists on my local box for some reason and so I want to ignore it.   As of today, we shouldn't check-in anything from that directory accidentally as that utility doesn't have any resource file associated with it.  
With that being said though, I am ok to remove this.

Comment (lenny-intel) [reply .gitignore:35]: Please remove. You shouldn't have one locally.

File: snap/snapcraft.yaml
Comment (tonyespy) [reply snap/snapcraft.yaml:155]: After discussing this afternoon, I think the original setting was incorrect, however for some reason didn't have a negative impact. As such, I'm marking this conversation resolved.

File: .gitignore
Comment (jim-wang-intel) [reply .gitignore]: removed.

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Review by tonyespy — DISMISSED** [review]
LGTM

**Comment by jim-wang-intel** [comment]
rebase and conflicts resolved.

**Comment by jim-wang-intel** [comment]
recheck

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**
